packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/AddAssetsToAlbum.kt

```diff
@@ -0,0 +1,88 @@
+package expo.modules.medialibrary.albums
+
+import android.content.Context
+import android.os.AsyncTask
+import android.provider.MediaStore
+import android.media.MediaScannerConnection
+import android.os.Build
+import expo.modules.core.Promise
+import expo.modules.medialibrary.MediaLibraryConstants.*
+import expo.modules.medialibrary.MediaLibraryUtils
+import java.io.File
+import java.io.IOException
+import java.util.concurrent.atomic.AtomicInteger
+
+internal class AddAssetsToAlbum(
+  private val context: Context,
+  private val assetIds: Array<String>,
+  private val albumId: String,
+  copyToAlbum: Boolean,
+  private val promise: Promise
+) : AsyncTask<Void?, Void?, Void?>() {
+  private val strategy = if (copyToAlbum) AssetFileStrategy.copyStrategy else AssetFileStrategy.moveStrategy
+
+  // Media store table can be corrupted. Extra check won't harm anyone.
+  private val album: File?
+    get() {
+      val path = arrayOf(MediaStore.MediaColumns.DATA)
+      val selection = "${MediaStore.MediaColumns.BUCKET_ID}=?"
+      val id = arrayOf(albumId)
+      context.contentResolver.query(
+        EXTERNAL_CONTENT,
+        path,
+        selection,
+        id,
+        null
+      ).use { albumCursor ->
+        if (albumCursor == null) {
+          promise.reject(ERROR_UNABLE_TO_LOAD, "Could not get album. Query returns null.")
+          return null
+        } else if (albumCursor.count == 0) {
+          promise.reject(ERROR_NO_ALBUM, "No album with id: $albumId")
+          return null
+        }
+        albumCursor.moveToNext()
+        val filePathColumnIndex = albumCursor.getColumnIndex(MediaStore.Images.Media.DATA)
+        val fileInAlbum = File(albumCursor.getString(filePathColumnIndex))
+
+        // Media store table can be corrupted. Extra check won't harm anyone.
+        if (!fileInAlbum.isFile) {
+          promise.reject(ERROR_MEDIA_LIBRARY_CORRUPTED, "Media library is corrupted")
+          return null
+        }
+        return File(fileInAlbum.parent!!)
```

Comment:
Must the parent of the `fileInAlbum` exist?

Reply:
When it is a file (if above), it must have a parent directory 😉